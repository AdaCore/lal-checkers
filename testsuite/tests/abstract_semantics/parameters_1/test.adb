procedure Test(a: in out Integer; b: in out Integer) is
   tmp : Integer;
begin
   tmp := a;
   a := b;
   b := tmp;
end Ex1;
