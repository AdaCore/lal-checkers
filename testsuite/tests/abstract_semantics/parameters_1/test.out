{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "a": "[-2147483648, 2147483647]", 
          "b": "[-2147483648, 2147483647]", 
          "tmp": "[-2147483648, 2147483647]"
        }
      }
    ]
  }
}
